tests/gui/test_widgets.py

```diff
@@ -2835,6 +2868,56 @@ def test_SpeechBubble__on_download_error(mocker):
     assert bubble.message.text() == error_message
 
 
+def test_CheckMark_eventFilter_hover(mocker):
+    mock_signal = mocker.MagicMock()
+    bubble = SpeechBubble(
+        "mock id",
+        "<b>hello</b>",
+        mock_signal,
+        mock_signal,
+        0,
+        SpeechBubble.MIN_CONTAINER_WIDTH,
+    )
+
+    bubble.check_mark = mocker.MagicMock()
+
+    # Hover enter
+    test_event = QEvent(QEvent.HoverEnter)
+    bubble.eventFilter(bubble, test_event)
+    assert bubble.check_mark.setIcon.call_count
+    bubble.check_mark.setIcon.reset_mock()  # ensure that it is the exact svg file we want
+
+    # Hover leave
+    test_event = QEvent(QEvent.HoverLeave)
+    bubble.eventFilter(bubble, test_event)
+    assert bubble.check_mark.setIcon.call_count == 1
+    bubble.check_mark.setIcon.reset_mock()
+
+
+def test_SpeechBubble_on_update_authenticated_user(mocker):
+    mock_update_signal = mocker.Mock()
+    mock_update_connect = mocker.Mock()
+    mock_update_signal.connect = mock_update_connect
+
+    mock_download_error_signal = mocker.Mock()
+    mock_download_error_connect = mocker.Mock()
+    mock_download_error_signal.connect = mock_download_error_connect
```

Comment:
Would it work if we turned this signal into a `MagicMock`?

```suggestion
    mock_download_error_signal = mocker.MagicMock()
```

I'd expect this to take care of the `connect` method as well.